src/config/__tests__/settings.spec.ts

```diff
@@ -0,0 +1,228 @@
+        version:                    '1.23.5',
+        memoryInGB:                 4,
+        numberCPUs:                 2,
+        port:                       6443,
+        containerEngine:            settings.ContainerEngine.MOBY,
+        checkForExistingKimBuilder: false,
+        enabled:                    true,
+        WSLIntegrations:            {},
+        options:                    {
+          traefik: true,
+          flannel: false,
+        },
+        suppressSudo: false
+      },
+      portForwarding: { includeKubernetesServices: false },
+      images:         {
+        showAll:   true,
+        namespace: 'k8s.io'
+      },
+      telemetry:              true,
+      updater:                true,
+      debug:                  true,
+      pathManagementStrategy: PathManagementStrategy.NotSet,
+    };
+  });
+
+  describe('getUpdatableNode', () => {
+    test('returns nil on an invalid top level accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'blah-blah-blah');
+
+      expect(result).toBeNull();
+    });
+    test('returns nil on an invalid internal accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes-options-blah');
+
+      expect(result).toBeNull();
+    });
+    test('returns the full pref with a top-level accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes');
+
+      if (!result) {
+        expect(result).not.toBeNull();
+      } else {
+        const lhs: RecursivePartial<settings.Settings> = result[0];
+        const accessor: string = result[1];
+
+        expect(lhs).toMatchObject(prefs);
+        expect(accessor).toBe('kubernetes');
+      }
+    });
+    test('returns a partial pref with an internal accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes-options-flannel');
+
+      if (!result) {
+        expect(result).not.toBeNull();
+      } else {
+        const lhs: RecursivePartial<settings.Settings> = result[0];
+        const accessor: string = result[1];
+        const flannelNow = prefs.kubernetes.options.flannel;
+        const flannelAfter = !flannelNow;
+
+        expect(lhs).toMatchObject({
+          traefik: prefs.kubernetes.options.traefik,
+          flannel: flannelNow,
+        });
+        expect(accessor).toBe('flannel');
+        (lhs as Record<string, any>)[accessor] = flannelAfter;
+        expect(prefs.kubernetes.options.flannel).toBe(flannelAfter);
+      }
+    });
+  });
+
+  test('no command-line args should leave prefs unchanged', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, []);
+
+    expect(newPrefs).toMatchObject(prefs);
+  });
+
+  test('one option with embedded equal sign should change only one value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-version=1.23.6']);
+
+    expect(newPrefs.kubernetes.version).toBe('1.23.6');
+    newPrefs.kubernetes.version = prefs.kubernetes.version;
```

Comment:
To double check things are you expect: add a `expect(newPrefs).not.toBeEqual(prefs);` before the assignment.

Reply:
Not needed with the deep copy.